kubernetes-azure-python/__main__.py

```diff
@@ -0,0 +1,120 @@
+"""An Azure RM Python Pulumi program"""
+
+import base64
+import pulumi
+from pulumi_azure_native import resources
+from pulumi_azure_native import network
+from pulumi_azure_native import containerservice
+
+# Get some values from the Pulumi configuration (or use defaults)
+azure_cfg = pulumi.Config("azure-native")
+azureLocation = azure_cfg.require("location")
```

Comment:
any specific reason you're flipping between `require()` and `get()` here?

Reply:
Yes. The `azure-native:location` configuration value _must_ be present (required by the provider as well as being used in the code), whereas the others have default values supplied if the configuration value isn't present. Is there a better/more recommended way of handling this sort of situation?

Reply:
Hm. I honestly don't know; what you're saying makes sense, but I've never used both in the same program, and I don't think the SDK docs make that clear. Maybe adjust the comment on line 9 to something like this?
```python
# Either pull required values from the Pulumi configuration, or get values and use defaults if null
```

Reply:
I'd overlooked another reason for using both `get()` and `require()` here. The last two project-namespaced configuration values (`mgmtGroupId` and `sshPubKey`) are values where I _can't_ provide a default value; they are too dependent on each customer's environment. I'll update the comments and add another one to better explain.

Reply:
Comments re-worded and a new comment added in 5b8b6ce

Reply:
This works, but since we panic when these two required values aren't provided (and there's no indication they're necessary until you run the `pulumi up` after the new-project wizard finishes), do you think it might be better to signal their requirement with config prompts, and just not supply default values?

Reply:
We can certainly add them to the `template` portion of `Pulumi.yaml` and not supply a default value. For the management group, I think that would probably work fine. However, I suspect the SSH public key will present a problem. You can't use `pulumi config set sshPubKey <key-material-here>` as it will throw an error. You have to use `pulumi config set sshPubKey < <path-to-public-key>`. (It has something to do with multi-line values, AIUI.) Let me try adding them to the `template` portion and see if there are errors with the SSH public key.

Reply:
Fixed in 270555d